unyt/_array_functions.py

```diff
@@ -714,3 +717,159 @@ def cumprod(a, *args, **kwargs):
+            )
+            * retu
+        )
+
+    res = np.choose._implementation(
+        a,
+        [np.asarray(c) for c in choices],
+        out=out.view(np.ndarray),
+        *args,
+        **kwargs,
+    )
+    if getattr(out, "units", None) is not None:
+        out.units = retu
+    return unyt_array(res, retu, bypass_validation=True)
+
+
+@implements(np.fill_diagonal)
+def fill_diagonal(a, val, *args, **kwargs) -> None:
+    _validate_units_consistency((a, val))
+    np.fill_diagonal._implementation(a.view(np.ndarray), val, *args, **kwargs)
+
+
+@implements(np.insert)
+def insert(arr, obj, values, *args, **kwargs):
+    retu = _validate_units_consistency((arr, values))
+    return (
+        np.insert._implementation(
+            arr.view(np.ndarray), obj, values.view(np.ndarray), *args, **kwargs
+        )
+        * retu
+    )
+
+
+@implements(np.isin)
+def isin(element, test_elements, *args, **kwargs):
+    _validate_units_consistency((element, test_elements))
+    return np.isin._implementation(
+        np.asarray(element), np.asarray(test_elements), *args, **kwargs
+    )
+
+
+@implements(np.place)
+def place(arr, mask, vals, *args, **kwargs) -> None:
+    _validate_units_consistency((arr, vals))
+    np.place._implementation(
+        arr.view(np.ndarray), mask, vals.view(np.ndarray), *args, **kwargs
+    )
+
+
+@implements(np.put)
+def put(a, ind, v, *args, **kwargs) -> None:
+    _validate_units_consistency((a, v))
+    np.put._implementation(a.view(np.ndarray), ind, v.view(np.ndarray))
+
+
+@implements(np.put_along_axis)
+def put_along_axis(arr, indices, values, axis, *args, **kwargs) -> None:
+    _validate_units_consistency((arr, values))
+    np.put_along_axis._implementation(
+        arr.view(np.ndarray), indices, values.view(np.ndarray), axis, *args, **kwargs
+    )
+
+
+@implements(np.putmask)
+def putmask(a, mask, values, *args, **kwargs) -> None:
+    _validate_units_consistency((a, values))
+    np.putmask._implementation(
+        a.view(np.ndarray), mask, values.view(np.ndarray), *args, **kwargs
+    )
+
+
+@implements(np.searchsorted)
+def searchsorted(a, v, *args, **kwargs):
+    _validate_units_consistency((a, v))
+    return np.searchsorted._implementation(
+        a.view(np.ndarray), np.asarray(v), *args, **kwargs
+    )
+
+
+@implements(np.select)
+def select(condlist, choicelist, default=0, *args, **kwargs):
+    retu = _validate_units_consistency((*choicelist, default))
+    return (
+        np.select._implementation(
+            condlist, [np.asarray(c) for c in choicelist], default
+        )
+        * retu
+    )
+
+
+@implements(np.setdiff1d)
+def setdiff1d(ar1, ar2, *args, **kwargs):
+    retu = _validate_units_consistency((ar1, ar2))
+    return (
+        np.setdiff1d._implementation(np.asarray(ar1), np.asarray(ar2), *args, **kwargs)
+        * retu
+    )
+
+
+@implements(np.where)
+def where(condition, x, y, *args, **kwargs):
```

Comment:
`x` and `y` are optional, so this will break someone doing `np.where(some_unyt_array)`.

I haven't carefully gone through elsewhere in the numpy API to check for whether optional arguments are handled correctly, it would be good to double-check that we're not making similar errors elsewhere in our wrapping.

Reply:
Good call, I didn't realise they were optional.

> it would be good to double-check that we're not making similar errors elsewhere in our wrapping.

agreed, however I am currently nowhere near the point where I'd feel like revisiting all of them.